docs/documentation/models/react.md

```diff
@@ -30,36 +30,100 @@ Hook for accessing LLM chat data and state.
 declare function useChat(options?: UseChatOptionsType): UseChatResultType;
 ```
 
-### Usage
+For a more detailed example, see the guide [Building client applications](/guides/models/building-client-applications.md).
+
+### Options
 
 ```ts
-const { input, setInput, messages, setMessages, loading, error, onChange, onSubmit } = useChat();
+/**
+ * The options supplied to the useChat hook.
+ */
+type UseChatOptionsType = {
+  /**
+   * The API endpoint to call when submitting a new message.
+   *
+   * Defaults to `/api/chat`.
+   */
+  url?: string;
+  /**
+   * Customize the request body sent to the API using this value. It accepts
+   * either a function or object.
+   *
+   * If given a function, the return value of the function will become the body
+   * of the request. The function will be passed the new message as its first
+   * argument and the history (an array of previous messages) as its second argument.
+   *
+   * If given an object, the object will be merged into the request body with the
+   * full set of messages, i.e., `{...body, messages }`.
+   *
+   * By default, the request body is `{ messages }`.
+   */
+  body?: BodyType;
+  /**
+   * Additional headers to send along with the request to the API.
+   */
+  headers?: Record<string, string>;
+  /**
+   * An accessor used to pluck out the message text. The response body or response
+   * stream can send back arbitrary values. If the value sent back is not the message
+   * text, then this component needs a way to access the message text. This function
+   * is given the value from the API as its input and should return the message text
+   * as its output.
+   *
+   * By default, it assumes the value from the API is the message text itself.
+   */
+  accessor?: (value: any) => string;
+  /**
+   * Initial message input. Defaults to empty string.
+   */
+  initialInput?: string;
```

Comment:
Nice. I was pushing the system message on the server because this didn't exist

Reply:
Maybe I'm misunderstanding you, but this is for the user's input and it already was a part of the library, I don't think it can be used for the system message.